### DISCUSSION TITLE
Using SPI1 instead of SPI

### DISCUSSION THREAD

**Discussion by EduanShuda** [sandeepmistry/arduino-LoRa#685] [Q&A]
Hi

I cannot get the standard library to work as it with SPI1. According to the API documentation:

### Set SPI interface

Override the default SPI interface used by the library. **Must** be called before `LoRa.begin()`.

```arduino
LoRa.setSPI(spi);
```
 * `spi` - new SPI interface to use, defaults to `SPI`

This call is optional and only needs to be used if you need to change the default SPI interface used, in the case your Arduino (or compatible) board has more than one SPI interface present.

I want to use SPI1.

In my sketch I have:
`#include <SPI1.h>`

In setup():
```
  LoRa.setSPI(SPI1);
  LoRa.setSPIFrequency(1E6);
  LoRa.setPins(SPI_CS, RST, IRQ_G0);

  if (!LoRa.begin(433900000)) {  // initialize ratio at 433900000 MHz
    while (true) {
      LED_FUNC(ERROR_LED, 5, 80);  // if failed, do nothing
    }
  }
```
  
  I get this error:
  Compilation error: no matching function for call to 'LoRaClass::setSPI(SPI1Class&)'

I have tried everything. Any help will be appreciated.

**Comment by morganrallen** [comment]
The error is a bit subtle but indicates the issue. The method `.setSPI` accepts `SPIClass& spi` and you're passing in a `SPI1Class`. I'm not sure where this class comes from (MiniCore?) but I don't believe it's part of standard Arduino, and it doesn't appear to inherit from `SPIClass`.  More info about your setup is needed to help further.

**Reply by EduanShuda** [reply]
I am using an ATmega328PB chipset.

The library I am using is SPI1.h 

I see that SPI1Class and SPI1Settings is used in the SPI library. The SPI1.h library came with the board files. 
Here is the link: https://github.com/watterott/ATmega328PB-Testing/tree/master/hardware/atmega328pb/avr/libraries

I always assumed this is the proper way to use SPI1 is by using SPI1.h
But there is no way is works with this LoRa library without changing loads of stuff in the library code.

Is the actual way to use SPI1 interface by calling SPI.h library? Why would they then have made SPI1.h?

Please advise on the proper way to use SPI1 (note that I am using SPI for other purposes so this also needs to be initialized)

Thanks

**Comment by paulfer747** [comment]
I would also like to know how to use SPI1 ?
